packages/account/src/Sections/Security/ClosingAccount/closing-account-reason-form.tsx

```diff
@@ -0,0 +1,266 @@
+        const final_reason = formatReasonsForCloseAccount(values);
+
+        onConfirmClick(final_reason);
+    };
+
+    const handleChangeCheckbox = (
+        values: FormikValues,
+        name: string,
+        setFieldValue: (name: string, values: string | boolean) => void
+    ) => {
+        if (!values[name]) {
+            dispatch({ type: SET_TOTAL_CHECKBOX_CHECKED, payload: total_checkbox_checked + 1 });
+
+            setFieldValue(name, !values[name]);
+        } else {
+            dispatch({ type: SET_TOTAL_CHECKBOX_CHECKED, payload: total_checkbox_checked - 1 });
+
+            setFieldValue(name, !values[name]);
+        }
+    };
+
+    const handleInputChange = (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        old_value: string,
+        onChange: (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void
+    ) => {
+        const value = event.target.value;
+        const is_delete_action = old_value.length > value.length;
+
+        if ((remaining_characters <= 0 || total_accumulated_characters >= CHARACTER_LIMIT) && !is_delete_action) {
+            event.preventDefault();
+        } else {
+            onChange(event);
+        }
+    };
+
+    const handleInputPaste = async (e: React.ClipboardEvent<HTMLInputElement | HTMLTextAreaElement>): Promise<void> => {
+        const clipboardData = e.clipboardData.getData('text') || (await navigator.clipboard.readText());
+
+        if (remaining_characters <= 0 || clipboardData.length > remaining_characters) {
+            e.preventDefault();
+        }
+    };
+
+    const formatReasonsForCloseAccount = (values: TClosingAccountFormValues) => {
+        let selected_reasons = selectedReasonsForCloseAccount(values)
+            .map(val => val[0])
+            .toString();
+        const is_other_trading_platform__has_value = !!values.other_trading_platforms.length;
+        const is_to_do_improve_has_value = !!values.do_to_improve.length;
+        if (is_other_trading_platform__has_value) {
+            selected_reasons = `${selected_reasons}, ${values.other_trading_platforms}`;
+        }
+        if (is_to_do_improve_has_value) {
+            selected_reasons = `${selected_reasons}, ${values.do_to_improve}`;
+        }
+
+        return selected_reasons.replace(/(\r\n|\n|\r)/gm, ' ');
+    };
+
+    const selectedReasonsForCloseAccount = (values: TClosingAccountFormValues) =>
+        Object.entries(values).filter(
+            ([key, value]) => !['other_trading_platforms', 'do_to_improve'].includes(key) && value
+        );
+
+    return (
+        <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={handleSubmitForm}>
+            {({ values, setFieldValue, errors, handleChange, dirty, isValid }) => (
+                <Form>
+                    {getCloseAccountReasonsList().map(reason => (
+                        <Field name={reason.name} key={reason.name}>
```

Comment:
As `Formik` supports handling checkboxes, we can refactor custom handler and move it to formik handler
ref: https://formik.org/docs/examples/checkboxes

Reply:
The usecase for our checkbox is different. we need to pass the values selected as a string